cmd/kubeadm/app/util/users/users_linux.go

```diff
@@ -629,3 +631,26 @@ func writeFile(f *os.File, str string) error {
 	}
 	return nil
 }
+
+// UpdateFileOwnerAndPermissions updates a files owner and permissions.
+func UpdateFileOwnerAndPermissions(path string, uid, gid int64, perms uint32) error {
+	if err := os.Chown(path, int(uid), int(gid)); err != nil {
+		return err
+	}
+	if err := os.Chmod(path, fs.FileMode(perms)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateDirOwnerAndPermissions updates the owner and permissions of all files
+// and sub-directories within the directory.
+func UpdateDirOwnerAndPermissions(dirPath string, uid, gid int64, perms uint32) error {
```

Comment:
i think we can export a single function:

`UpdatePathOwnerAndPermissions()`


inside the function there should be a check if the path is a directory.

Reply:
use `UpdatePathOwnerAndPermissions` instead of `UpdateDirOwnerAndPermissions` as the function name to match the comment.

Reply:
Oops! fixed.